admin/tools/docker/replication/deployment/ncsa/env.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# This file is part of lsst/qserv.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# This script is supposed to be sourced from a client script in order
+# to set up proper values of the corresponding parameters
+
+basedir=$(dirname $0)
+if [ -z "$basedir" ] || [ "$0" = "bash" ]; then
+    (>&2 echo "error: variable 'basedir' is not defined")
```

Comment:
BTW, why do you need to run `echo` in a sub-shell? Forking for just executing one builtin command looks a bit expensive

Reply:
This is probably an artifact of the original code when this line was inside a function. So I just wanted to redirect the error message into the standard error string to avoid returning it as a result of that function. I will revisit this code again to see if I should get rid of it

Reply:
Fixed